controllers/article.js

```diff
@@ -99,56 +99,46 @@ export default {
 
   async getArticles(req, res) {
     const {
-      category,
-      author,
-      tag,
-      keywords
+      category, author, tag, keywords, limit
     } = req.query;
-    const queryArray = [
-      {
-        [or]: {
-          title: { [iLike]: keywords ? `%${keywords}%` : '%' },
-          description: { [iLike]: keywords ? `%${keywords}%` : '%' },
-          body: { [iLike]: keywords ? `%${keywords}%` : '%' }
-        },
-      },
-      {
-        category: category || categories
+    let offset, { page } = req.query;
+    const queryArray = [{
+      [or]: {
+        title: { [iLike]: keywords ? `%${keywords}%` : '%' },
+        description: { [iLike]: keywords ? `%${keywords}%` : '%' },
+        body: { [iLike]: keywords ? `%${keywords}%` : '%' }
       },
-      {
-        tags: tag ? { [contains]: [tag] } : { [notIn]: [] }
-      }
-    ];
+    },
+    { category: category || categories },
+    { tags: tag ? { [contains]: [tag] } : { [notIn]: [] } }];
     if (author) {
       queryArray.push({
         [or]: {
-          '$User.Profile.lastName$': {
-            [iLike]: `%${author}%`
-          },
-          '$User.Profile.firstName$': {
-            [iLike]: `%${author}%`
-          },
-          '$User.username$': {
-            [iLike]: `%${author}%`
-          },
+          '$User.Profile.lastName$': { [iLike]: `%${author}%` },
+          '$User.Profile.firstName$': { [iLike]: `%${author}%` },
+          '$User.username$': { [iLike]: `%${author}%` },
         }
-
       });
     }
+    offset = limit * (page - 1);
```

Comment:
What inspired this computation?

Reply:
The `offset` skips a particular number of roles in the database, while the `limit` represents the number of roles to be returned from the database.
So if `page number` is `1`, and `limit` is `5` then the number of roles to be skipped would be `5 * (1-1)` which is `0`. With this, our `db` returns the first 5 rows as page 1.